src/libtiled/mapreader.cpp

```diff
@@ -964,7 +965,12 @@ void MapReaderPrivate::readProperty(Properties *properties)
         }
     }
 
-    properties->insert(propertyName, propertyValue);
+    if(propertyType.isEmpty() || propertyType.isNull()){
```

Comment:
`QString::isNull` implies `QString::isEmpty` so you can drop the second condition.